### PR TITLE
[RO-82] Line Item listing endpoint 

### PR DESCRIPTION
**Pull request by luisfn**
Ticket: https://oat-sa.atlassian.net/browse/RO-82

Purpose of this PR is to add a new endpoint to list line items.

### CODE REVIEW

**Review by uncleempty — DISMISSED** [review]
Good one, thanks!
Please, don't forget to update changelog
**Review Checklist**
- [X] New code is covered by tests (if applicable)
- [X] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [X] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [X] Documentation is updated according to changes (if applicable)

File: config/packages/security.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ security:
 
         api_key:
             anonymous: ~
-            pattern: '^%app.route_prefix%/v1/bulk'
+            pattern: '^%app.route_prefix%/v1/(bulk|line-items)'
```
Comment (andreluizmachado) [inline config/packages/security.yaml:29]: Just do not forget to notify the applications who are using this endpoint that it was changed :)

File: src/Action/LineItem/ListLineItemsAction.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        $this->lineItemFindCriteriaFactory = $lineItemFindCriteriaFactory;
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        $findLineItemCriteria = $this->lineItemFindCriteriaFactory->create($request);
+
+        $cursor = $request->get('cursor') ? (int) $request->get('cursor') : null;
+        $limit = ($request->get('limit') === null
+            || (int) $request->get('limit') > LineItemRepository::MAX_LINE_ITEM_LIMIT)
```
Comment (andreluizmachado) [inline src/Action/LineItem/ListLineItemsAction.php:54]: Suggestion: I'd return an error in case the limit is greater than the maximum

**Review by andreluizmachado — COMMENTED** [review]
Amazing job @luisfn. I just left 2 comments.

File: src/Action/LineItem/ListLineItemsAction.php (hunk 2 above)
Comment (luisfn) [reply src/Action/LineItem/ListLineItemsAction.php]: Fixed

**Review by andreluizmachado — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Changelog is updated according to changes (if applicable)
- [x] Documentation is updated according to changes (if applicable)

**Review by uncleempty — APPROVED** [review]
**Review Checklist**
- [X] New code is covered by tests (if applicable)
- [X] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [X] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful
- [X] Changelog is updated according to changes (if applicable)
- [X] Documentation is updated according to changes (if applicable)

**Pull request merged**